File: Programming EF Code First/chapter3/Chapter3/BAGA_DataAnnotations/Model/Trip.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Model
{
  public class Trip
  {
    [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Identifier { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public decimal CostUSD { get; set; }
    [Timestamp]
    public byte[] RowVersion { get; set; }
  }
}

File: Programming EF Code First/chapter3/Chapter 3 VB/Data Annotations/Model/Trip.vb
Imports System.ComponentModel.DataAnnotations

Namespace Model
  Public Class Trip
	<Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)>
	Public Property Identifier() As Guid
	Public Property StartDate() As Date
	Public Property EndDate() As Date
	Public Property CostUSD() As Decimal
	<Timestamp>
	Public Property RowVersion() As Byte()
  End Class
End Namespace